{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting scikit-bio\n",
      "  Downloading scikit-bio-0.6.2.tar.gz (2.7 MB)\n",
      "     ---------------------------------------- 0.0/2.7 MB ? eta -:--:--\n",
      "     ----------- ---------------------------- 0.8/2.7 MB 6.6 MB/s eta 0:00:01\n",
      "     ---------------------------------------- 2.7/2.7 MB 8.6 MB/s eta 0:00:00\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'done'\n",
      "  Preparing metadata (pyproject.toml): started\n",
      "  Preparing metadata (pyproject.toml): finished with status 'done'\n",
      "Requirement already satisfied: requests>=2.20.0 in c:\\users\\utente\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from scikit-bio) (2.32.3)\n",
      "Requirement already satisfied: decorator>=3.4.2 in c:\\users\\utente\\appdata\\roaming\\python\\python311\\site-packages (from scikit-bio) (5.1.1)\n",
      "Collecting natsort>=4.0.3 (from scikit-bio)\n",
      "  Downloading natsort-8.4.0-py3-none-any.whl.metadata (21 kB)\n",
      "Requirement already satisfied: numpy>=1.17.0 in c:\\users\\utente\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from scikit-bio) (1.26.4)\n",
      "Requirement already satisfied: pandas>=1.5.0 in c:\\users\\utente\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from scikit-bio) (2.2.3)\n",
      "Requirement already satisfied: scipy>=1.9.0 in c:\\users\\utente\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from scikit-bio) (1.14.1)\n",
      "Collecting h5py>=3.6.0 (from scikit-bio)\n",
      "  Using cached h5py-3.12.1-cp311-cp311-win_amd64.whl.metadata (2.5 kB)\n",
      "Collecting biom-format>=2.1.16 (from scikit-bio)\n",
      "  Downloading biom-format-2.1.16.tar.gz (11.7 MB)\n",
      "     ---------------------------------------- 0.0/11.7 MB ? eta -:--:--\n",
      "     - -------------------------------------- 0.5/11.7 MB 3.3 MB/s eta 0:00:04\n",
      "     ------- -------------------------------- 2.1/11.7 MB 6.2 MB/s eta 0:00:02\n",
      "     ---------- ----------------------------- 3.1/11.7 MB 5.9 MB/s eta 0:00:02\n",
      "     ---------------- ----------------------- 4.7/11.7 MB 6.2 MB/s eta 0:00:02\n",
      "     -------------------- ------------------- 6.0/11.7 MB 6.1 MB/s eta 0:00:01\n",
      "     --------------------------- ------------ 8.1/11.7 MB 6.8 MB/s eta 0:00:01\n",
      "     --------------------------------- ------ 10.0/11.7 MB 7.1 MB/s eta 0:00:01\n",
      "     ---------------------------------------  11.5/11.7 MB 7.4 MB/s eta 0:00:01\n",
      "     ---------------------------------------  11.5/11.7 MB 7.4 MB/s eta 0:00:01\n",
      "     ---------------------------------------- 11.7/11.7 MB 5.8 MB/s eta 0:00:00\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'done'\n",
      "  Preparing metadata (pyproject.toml): started\n",
      "  Preparing metadata (pyproject.toml): finished with status 'done'\n",
      "Collecting statsmodels>=0.14.0 (from scikit-bio)\n",
      "  Downloading statsmodels-0.14.4-cp311-cp311-win_amd64.whl.metadata (9.5 kB)\n",
      "Requirement already satisfied: click in c:\\users\\utente\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from biom-format>=2.1.16->scikit-bio) (8.1.7)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in c:\\users\\utente\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from pandas>=1.5.0->scikit-bio) (2.9.0.post0)\n",
      "Requirement already satisfied: pytz>=2020.1 in c:\\users\\utente\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from pandas>=1.5.0->scikit-bio) (2024.2)\n",
      "Requirement already satisfied: tzdata>=2022.7 in c:\\users\\utente\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from pandas>=1.5.0->scikit-bio) (2024.2)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in c:\\users\\utente\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from requests>=2.20.0->scikit-bio) (3.4.0)\n",
      "Requirement already satisfied: idna<4,>=2.5 in c:\\users\\utente\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from requests>=2.20.0->scikit-bio) (3.10)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in c:\\users\\utente\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from requests>=2.20.0->scikit-bio) (2.2.3)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\users\\utente\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from requests>=2.20.0->scikit-bio) (2024.8.30)\n",
      "Collecting patsy>=0.5.6 (from statsmodels>=0.14.0->scikit-bio)\n",
      "  Downloading patsy-1.0.1-py2.py3-none-any.whl.metadata (3.3 kB)\n",
      "Requirement already satisfied: packaging>=21.3 in c:\\users\\utente\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from statsmodels>=0.14.0->scikit-bio) (24.2)\n",
      "Requirement already satisfied: six>=1.5 in c:\\users\\utente\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from python-dateutil>=2.8.2->pandas>=1.5.0->scikit-bio) (1.17.0)\n",
      "Requirement already satisfied: colorama in c:\\users\\utente\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from click->biom-format>=2.1.16->scikit-bio) (0.4.6)\n",
      "Using cached h5py-3.12.1-cp311-cp311-win_amd64.whl (3.0 MB)\n",
      "Downloading natsort-8.4.0-py3-none-any.whl (38 kB)\n",
      "Downloading statsmodels-0.14.4-cp311-cp311-win_amd64.whl (9.9 MB)\n",
      "   ---------------------------------------- 0.0/9.9 MB ? eta -:--:--\n",
      "   ---- ----------------------------------- 1.0/9.9 MB 5.6 MB/s eta 0:00:02\n",
      "   ---------- ----------------------------- 2.6/9.9 MB 6.6 MB/s eta 0:00:02\n",
      "   --------------- ------------------------ 3.9/9.9 MB 6.5 MB/s eta 0:00:01\n",
      "   ----------------- ---------------------- 4.2/9.9 MB 6.8 MB/s eta 0:00:01\n",
      "   ------------------ --------------------- 4.5/9.9 MB 4.5 MB/s eta 0:00:02\n",
      "   ---------------------- ----------------- 5.5/9.9 MB 4.5 MB/s eta 0:00:01\n",
      "   ------------------------- -------------- 6.3/9.9 MB 4.3 MB/s eta 0:00:01\n",
      "   ----------------------------- ---------- 7.3/9.9 MB 4.4 MB/s eta 0:00:01\n",
      "   ----------------------------------- ---- 8.7/9.9 MB 4.6 MB/s eta 0:00:01\n",
      "   ---------------------------------------  9.7/9.9 MB 4.6 MB/s eta 0:00:01\n",
      "   ---------------------------------------  9.7/9.9 MB 4.6 MB/s eta 0:00:01\n",
      "   ---------------------------------------  9.7/9.9 MB 4.6 MB/s eta 0:00:01\n",
      "   ---------------------------------------  9.7/9.9 MB 4.6 MB/s eta 0:00:01\n",
      "   ---------------------------------------- 9.9/9.9 MB 3.4 MB/s eta 0:00:00\n",
      "Downloading patsy-1.0.1-py2.py3-none-any.whl (232 kB)\n",
      "Building wheels for collected packages: scikit-bio, biom-format\n",
      "  Building wheel for scikit-bio (pyproject.toml): started\n",
      "  Building wheel for scikit-bio (pyproject.toml): finished with status 'error'\n",
      "  Building wheel for biom-format (pyproject.toml): started\n",
      "  Building wheel for biom-format (pyproject.toml): finished with status 'error'\n",
      "Failed to build scikit-bio biom-format\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  error: subprocess-exited-with-error\n",
      "  \n",
      "  × Building wheel for scikit-bio (pyproject.toml) did not run successfully.\n",
      "  │ exit code: 1\n",
      "  ╰─> [819 lines of output]\n",
      "      [1/5] Cythonizing skbio/metadata/_intersection.pyx\n",
      "      C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-1449x05t\\overlay\\Lib\\site-packages\\Cython\\Compiler\\Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-install-_c7v_ck_\\scikit-bio_0237d7107eda42d4be0e097b691cf5ad\\skbio\\metadata\\_intersection.pyx\n",
      "        tree = Parsing.p_module(s, pxd, full_module_name)\n",
      "      [2/5] Cythonizing skbio/stats/ordination/_cutils.pyx\n",
      "      C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-1449x05t\\overlay\\Lib\\site-packages\\Cython\\Compiler\\Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-install-_c7v_ck_\\scikit-bio_0237d7107eda42d4be0e097b691cf5ad\\skbio\\stats\\ordination\\_cutils.pyx\n",
      "        tree = Parsing.p_module(s, pxd, full_module_name)\n",
      "      [3/5] Cythonizing skbio/alignment/_ssw_wrapper.pyx\n",
      "      C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-1449x05t\\overlay\\Lib\\site-packages\\Cython\\Compiler\\Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-install-_c7v_ck_\\scikit-bio_0237d7107eda42d4be0e097b691cf5ad\\skbio\\alignment\\_ssw_wrapper.pyx\n",
      "        tree = Parsing.p_module(s, pxd, full_module_name)\n",
      "      [4/5] Cythonizing skbio/diversity/_phylogenetic.pyx\n",
      "      C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-1449x05t\\overlay\\Lib\\site-packages\\Cython\\Compiler\\Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-install-_c7v_ck_\\scikit-bio_0237d7107eda42d4be0e097b691cf5ad\\skbio\\diversity\\_phylogenetic.pyx\n",
      "        tree = Parsing.p_module(s, pxd, full_module_name)\n",
      "      [5/5] Cythonizing skbio/stats/distance/_cutils.pyx\n",
      "      C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-1449x05t\\overlay\\Lib\\site-packages\\Cython\\Compiler\\Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-install-_c7v_ck_\\scikit-bio_0237d7107eda42d4be0e097b691cf5ad\\skbio\\stats\\distance\\_cutils.pyx\n",
      "        tree = Parsing.p_module(s, pxd, full_module_name)\n",
      "      performance hint: skbio\\stats\\distance\\_cutils.pyx:398:5: Exception check on '_dist_euclidean' will always require the GIL to be acquired. Declare '_dist_euclidean' as 'noexcept' if you control the definition and you're sure you don't want the function to raise exceptions.\n",
      "      performance hint: skbio\\stats\\distance\\_cutils.pyx:407:5: Exception check on '_norm_euclidean' will always require the GIL to be acquired. Declare '_norm_euclidean' as 'noexcept' if you control the definition and you're sure you don't want the function to raise exceptions.\n",
      "      performance hint: skbio\\stats\\distance\\_cutils.pyx:414:5: Exception check on '_sum' will always require the GIL to be acquired. Declare '_sum' as 'noexcept' if you control the definition and you're sure you don't want the function to raise exceptions.\n",
      "      performance hint: skbio\\stats\\distance\\_cutils.pyx:350:36: Exception check after calling '__pyx_fuse_0_dist_euclidean' will always require the GIL to be acquired. Declare '__pyx_fuse_0_dist_euclidean' as 'noexcept' if you control the definition and you're sure you don't want the function to raise exceptions.\n",
      "      performance hint: skbio\\stats\\distance\\_cutils.pyx:357:24: Exception check after calling '__pyx_fuse_0_sum' will always require the GIL to be acquired. Declare '__pyx_fuse_0_sum' as 'noexcept' if you control the definition and you're sure you don't want the function to raise exceptions.\n",
      "      performance hint: skbio\\stats\\distance\\_cutils.pyx:381:35: Exception check after calling '__pyx_fuse_0_norm_euclidean' will always require the GIL to be acquired. Declare '__pyx_fuse_0_norm_euclidean' as 'noexcept' if you control the definition and you're sure you don't want the function to raise exceptions.\n",
      "      performance hint: skbio\\stats\\distance\\_cutils.pyx:389:34: Exception check after calling '__pyx_fuse_0_dist_euclidean' will always require the GIL to be acquired. Declare '__pyx_fuse_0_dist_euclidean' as 'noexcept' if you control the definition and you're sure you don't want the function to raise exceptions.\n",
      "      performance hint: skbio\\stats\\distance\\_cutils.pyx:350:36: Exception check after calling '__pyx_fuse_1_dist_euclidean' will always require the GIL to be acquired. Declare '__pyx_fuse_1_dist_euclidean' as 'noexcept' if you control the definition and you're sure you don't want the function to raise exceptions.\n",
      "      performance hint: skbio\\stats\\distance\\_cutils.pyx:357:24: Exception check after calling '__pyx_fuse_1_sum' will always require the GIL to be acquired. Declare '__pyx_fuse_1_sum' as 'noexcept' if you control the definition and you're sure you don't want the function to raise exceptions.\n",
      "      performance hint: skbio\\stats\\distance\\_cutils.pyx:381:35: Exception check after calling '__pyx_fuse_1_norm_euclidean' will always require the GIL to be acquired. Declare '__pyx_fuse_1_norm_euclidean' as 'noexcept' if you control the definition and you're sure you don't want the function to raise exceptions.\n",
      "      performance hint: skbio\\stats\\distance\\_cutils.pyx:389:34: Exception check after calling '__pyx_fuse_1_dist_euclidean' will always require the GIL to be acquired. Declare '__pyx_fuse_1_dist_euclidean' as 'noexcept' if you control the definition and you're sure you don't want the function to raise exceptions.\n",
      "      warning: skbio\\stats\\distance\\_cutils.pyx:403:15: Use boundscheck(False) for faster access\n",
      "      warning: skbio\\stats\\distance\\_cutils.pyx:403:22: Use boundscheck(False) for faster access\n",
      "      warning: skbio\\stats\\distance\\_cutils.pyx:403:15: Use boundscheck(False) for faster access\n",
      "      warning: skbio\\stats\\distance\\_cutils.pyx:403:22: Use boundscheck(False) for faster access\n",
      "      warning: skbio\\stats\\distance\\_cutils.pyx:411:14: Use boundscheck(False) for faster access\n",
      "      warning: skbio\\stats\\distance\\_cutils.pyx:411:21: Use boundscheck(False) for faster access\n",
      "      warning: skbio\\stats\\distance\\_cutils.pyx:411:14: Use boundscheck(False) for faster access\n",
      "      warning: skbio\\stats\\distance\\_cutils.pyx:411:21: Use boundscheck(False) for faster access\n",
      "      warning: skbio\\stats\\distance\\_cutils.pyx:418:18: Use boundscheck(False) for faster access\n",
      "      warning: skbio\\stats\\distance\\_cutils.pyx:418:18: Use boundscheck(False) for faster access\n",
      "      C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-1449x05t\\overlay\\Lib\\site-packages\\setuptools\\_distutils\\dist.py:261: UserWarning: Unknown distribution option: 'tests_require'\n",
      "        warnings.warn(msg)\n",
      "      running bdist_wheel\n",
      "      running build\n",
      "      running build_py\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\n",
      "      copying skbio\\test.py -> build\\lib.win-amd64-cpython-311\\skbio\n",
      "      copying skbio\\workflow.py -> build\\lib.win-amd64-cpython-311\\skbio\n",
      "      copying skbio\\_base.py -> build\\lib.win-amd64-cpython-311\\skbio\n",
      "      copying skbio\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\alignment\n",
      "      copying skbio\\alignment\\_indexing.py -> build\\lib.win-amd64-cpython-311\\skbio\\alignment\n",
      "      copying skbio\\alignment\\_pairwise.py -> build\\lib.win-amd64-cpython-311\\skbio\\alignment\n",
      "      copying skbio\\alignment\\_path.py -> build\\lib.win-amd64-cpython-311\\skbio\\alignment\n",
      "      copying skbio\\alignment\\_repr.py -> build\\lib.win-amd64-cpython-311\\skbio\\alignment\n",
      "      copying skbio\\alignment\\_tabular_msa.py -> build\\lib.win-amd64-cpython-311\\skbio\\alignment\n",
      "      copying skbio\\alignment\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\alignment\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\diversity\n",
      "      copying skbio\\diversity\\_block.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\n",
      "      copying skbio\\diversity\\_driver.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\n",
      "      copying skbio\\diversity\\_util.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\n",
      "      copying skbio\\diversity\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\embedding\n",
      "      copying skbio\\embedding\\_embedding.py -> build\\lib.win-amd64-cpython-311\\skbio\\embedding\n",
      "      copying skbio\\embedding\\_protein.py -> build\\lib.win-amd64-cpython-311\\skbio\\embedding\n",
      "      copying skbio\\embedding\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\embedding\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\io\n",
      "      copying skbio\\io\\registry.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\n",
      "      copying skbio\\io\\util.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\n",
      "      copying skbio\\io\\_exception.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\n",
      "      copying skbio\\io\\_fileobject.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\n",
      "      copying skbio\\io\\_iosources.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\n",
      "      copying skbio\\io\\_warning.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\n",
      "      copying skbio\\io\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\metadata\n",
      "      copying skbio\\metadata\\base.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\n",
      "      copying skbio\\metadata\\io.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\n",
      "      copying skbio\\metadata\\missing.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\n",
      "      copying skbio\\metadata\\_enan.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\n",
      "      copying skbio\\metadata\\_interval.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\n",
      "      copying skbio\\metadata\\_metadata.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\n",
      "      copying skbio\\metadata\\_mixin.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\n",
      "      copying skbio\\metadata\\_repr.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\n",
      "      copying skbio\\metadata\\_testing.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\n",
      "      copying skbio\\metadata\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\sequence\n",
      "      copying skbio\\sequence\\distance.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\n",
      "      copying skbio\\sequence\\_alphabet.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\n",
      "      copying skbio\\sequence\\_dna.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\n",
      "      copying skbio\\sequence\\_genetic_code.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\n",
      "      copying skbio\\sequence\\_grammared_sequence.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\n",
      "      copying skbio\\sequence\\_nucleotide_mixin.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\n",
      "      copying skbio\\sequence\\_protein.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\n",
      "      copying skbio\\sequence\\_repr.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\n",
      "      copying skbio\\sequence\\_rna.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\n",
      "      copying skbio\\sequence\\_sequence.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\n",
      "      copying skbio\\sequence\\_substitution.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\n",
      "      copying skbio\\sequence\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\stats\n",
      "      copying skbio\\stats\\composition.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\n",
      "      copying skbio\\stats\\gradient.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\n",
      "      copying skbio\\stats\\power.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\n",
      "      copying skbio\\stats\\_misc.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\n",
      "      copying skbio\\stats\\_subsample.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\n",
      "      copying skbio\\stats\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\table\n",
      "      copying skbio\\table\\_base.py -> build\\lib.win-amd64-cpython-311\\skbio\\table\n",
      "      copying skbio\\table\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\table\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\tests\n",
      "      copying skbio\\tests\\test_base.py -> build\\lib.win-amd64-cpython-311\\skbio\\tests\n",
      "      copying skbio\\tests\\test_workflow.py -> build\\lib.win-amd64-cpython-311\\skbio\\tests\n",
      "      copying skbio\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\tree\n",
      "      copying skbio\\tree\\_exception.py -> build\\lib.win-amd64-cpython-311\\skbio\\tree\n",
      "      copying skbio\\tree\\_majority_rule.py -> build\\lib.win-amd64-cpython-311\\skbio\\tree\n",
      "      copying skbio\\tree\\_nj.py -> build\\lib.win-amd64-cpython-311\\skbio\\tree\n",
      "      copying skbio\\tree\\_tree.py -> build\\lib.win-amd64-cpython-311\\skbio\\tree\n",
      "      copying skbio\\tree\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\tree\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\util\n",
      "      copying skbio\\util\\_decorator.py -> build\\lib.win-amd64-cpython-311\\skbio\\util\n",
      "      copying skbio\\util\\_exception.py -> build\\lib.win-amd64-cpython-311\\skbio\\util\n",
      "      copying skbio\\util\\_misc.py -> build\\lib.win-amd64-cpython-311\\skbio\\util\n",
      "      copying skbio\\util\\_plotting.py -> build\\lib.win-amd64-cpython-311\\skbio\\util\n",
      "      copying skbio\\util\\_testing.py -> build\\lib.win-amd64-cpython-311\\skbio\\util\n",
      "      copying skbio\\util\\_warning.py -> build\\lib.win-amd64-cpython-311\\skbio\\util\n",
      "      copying skbio\\util\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\util\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\alignment\\tests\n",
      "      copying skbio\\alignment\\tests\\test_pairwise.py -> build\\lib.win-amd64-cpython-311\\skbio\\alignment\\tests\n",
      "      copying skbio\\alignment\\tests\\test_path.py -> build\\lib.win-amd64-cpython-311\\skbio\\alignment\\tests\n",
      "      copying skbio\\alignment\\tests\\test_ssw.py -> build\\lib.win-amd64-cpython-311\\skbio\\alignment\\tests\n",
      "      copying skbio\\alignment\\tests\\test_tabular_msa.py -> build\\lib.win-amd64-cpython-311\\skbio\\alignment\\tests\n",
      "      copying skbio\\alignment\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\alignment\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\alignment\\_lib\n",
      "      copying skbio\\alignment\\_lib\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\alignment\\_lib\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\n",
      "      copying skbio\\diversity\\alpha\\_ace.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\n",
      "      copying skbio\\diversity\\alpha\\_base.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\n",
      "      copying skbio\\diversity\\alpha\\_chao1.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\n",
      "      copying skbio\\diversity\\alpha\\_gini.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\n",
      "      copying skbio\\diversity\\alpha\\_lladser.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\n",
      "      copying skbio\\diversity\\alpha\\_pd.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\n",
      "      copying skbio\\diversity\\alpha\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\diversity\\beta\n",
      "      copying skbio\\diversity\\beta\\_unifrac.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\beta\n",
      "      copying skbio\\diversity\\beta\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\beta\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\diversity\\tests\n",
      "      copying skbio\\diversity\\tests\\test_block.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\tests\n",
      "      copying skbio\\diversity\\tests\\test_driver.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\tests\n",
      "      copying skbio\\diversity\\tests\\test_util.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\tests\n",
      "      copying skbio\\diversity\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\\tests\n",
      "      copying skbio\\diversity\\alpha\\tests\\test_ace.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\\tests\n",
      "      copying skbio\\diversity\\alpha\\tests\\test_base.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\\tests\n",
      "      copying skbio\\diversity\\alpha\\tests\\test_chao1.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\\tests\n",
      "      copying skbio\\diversity\\alpha\\tests\\test_gini.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\\tests\n",
      "      copying skbio\\diversity\\alpha\\tests\\test_lladser.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\\tests\n",
      "      copying skbio\\diversity\\alpha\\tests\\test_pd.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\\tests\n",
      "      copying skbio\\diversity\\alpha\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\diversity\\beta\\tests\n",
      "      copying skbio\\diversity\\beta\\tests\\test_unifrac.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\beta\\tests\n",
      "      copying skbio\\diversity\\beta\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\beta\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\embedding\\tests\n",
      "      copying skbio\\embedding\\tests\\test_embedding.py -> build\\lib.win-amd64-cpython-311\\skbio\\embedding\\tests\n",
      "      copying skbio\\embedding\\tests\\test_protein.py -> build\\lib.win-amd64-cpython-311\\skbio\\embedding\\tests\n",
      "      copying skbio\\embedding\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\embedding\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\binary_dm.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\biom.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\blast6.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\blast7.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\clustal.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\embed.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\embl.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\emptyfile.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\fasta.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\fastq.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\genbank.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\gff3.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\lsmat.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\newick.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\ordination.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\phylip.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\qseq.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\sample_metadata.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\stockholm.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\taxdump.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\_base.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\_blast.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\_sequence_feature_vocabulary.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      copying skbio\\io\\format\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\io\\tests\n",
      "      copying skbio\\io\\tests\\test_iosources.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\tests\n",
      "      copying skbio\\io\\tests\\test_registry.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\tests\n",
      "      copying skbio\\io\\tests\\test_util.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\tests\n",
      "      copying skbio\\io\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_base.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_binary_dm.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_biom.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_blast6.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_blast7.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_clustal.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_embed.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_embl.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_emptyfile.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_fasta.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_fastq.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_genbank.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_gff3.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_lsmat.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_newick.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_ordination.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_phylip.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_qseq.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_sample_metadata.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_sequence_feature_vocabulary.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_stockholm.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\test_taxdump.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      copying skbio\\io\\format\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\n",
      "      copying skbio\\metadata\\tests\\test_enan.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\n",
      "      copying skbio\\metadata\\tests\\test_intersection.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\n",
      "      copying skbio\\metadata\\tests\\test_interval.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\n",
      "      copying skbio\\metadata\\tests\\test_io.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\n",
      "      copying skbio\\metadata\\tests\\test_metadata.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\n",
      "      copying skbio\\metadata\\tests\\test_metadata_column.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\n",
      "      copying skbio\\metadata\\tests\\test_missing.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\n",
      "      copying skbio\\metadata\\tests\\test_mixin.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\n",
      "      copying skbio\\metadata\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\sequence\\tests\n",
      "      copying skbio\\sequence\\tests\\test_alphabet.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\\tests\n",
      "      copying skbio\\sequence\\tests\\test_distance.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\\tests\n",
      "      copying skbio\\sequence\\tests\\test_dna.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\\tests\n",
      "      copying skbio\\sequence\\tests\\test_genetic_code.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\\tests\n",
      "      copying skbio\\sequence\\tests\\test_grammared_sequence.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\\tests\n",
      "      copying skbio\\sequence\\tests\\test_nucleotide_sequences.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\\tests\n",
      "      copying skbio\\sequence\\tests\\test_protein.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\\tests\n",
      "      copying skbio\\sequence\\tests\\test_rna.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\\tests\n",
      "      copying skbio\\sequence\\tests\\test_sequence.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\\tests\n",
      "      copying skbio\\sequence\\tests\\test_substitution.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\\tests\n",
      "      copying skbio\\sequence\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\sequence\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\n",
      "      copying skbio\\stats\\distance\\_anosim.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\n",
      "      copying skbio\\stats\\distance\\_base.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\n",
      "      copying skbio\\stats\\distance\\_bioenv.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\n",
      "      copying skbio\\stats\\distance\\_mantel.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\n",
      "      copying skbio\\stats\\distance\\_permanova.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\n",
      "      copying skbio\\stats\\distance\\_permdisp.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\n",
      "      copying skbio\\stats\\distance\\_utils.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\n",
      "      copying skbio\\stats\\distance\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\stats\\evolve\n",
      "      copying skbio\\stats\\evolve\\_hommola.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\evolve\n",
      "      copying skbio\\stats\\evolve\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\evolve\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\n",
      "      copying skbio\\stats\\ordination\\_canonical_correspondence_analysis.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\n",
      "      copying skbio\\stats\\ordination\\_correspondence_analysis.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\n",
      "      copying skbio\\stats\\ordination\\_ordination_results.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\n",
      "      copying skbio\\stats\\ordination\\_principal_coordinate_analysis.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\n",
      "      copying skbio\\stats\\ordination\\_redundancy_analysis.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\n",
      "      copying skbio\\stats\\ordination\\_utils.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\n",
      "      copying skbio\\stats\\ordination\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\n",
      "      copying skbio\\stats\\tests\\test_composition.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\n",
      "      copying skbio\\stats\\tests\\test_gradient.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\n",
      "      copying skbio\\stats\\tests\\test_misc.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\n",
      "      copying skbio\\stats\\tests\\test_power.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\n",
      "      copying skbio\\stats\\tests\\test_subsample.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\n",
      "      copying skbio\\stats\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\n",
      "      copying skbio\\stats\\distance\\tests\\test_anosim.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\n",
      "      copying skbio\\stats\\distance\\tests\\test_base.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\n",
      "      copying skbio\\stats\\distance\\tests\\test_bioenv.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\n",
      "      copying skbio\\stats\\distance\\tests\\test_mantel.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\n",
      "      copying skbio\\stats\\distance\\tests\\test_permanova.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\n",
      "      copying skbio\\stats\\distance\\tests\\test_permdisp.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\n",
      "      copying skbio\\stats\\distance\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\stats\\evolve\\tests\n",
      "      copying skbio\\stats\\evolve\\tests\\test_hommola.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\evolve\\tests\n",
      "      copying skbio\\stats\\evolve\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\evolve\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\n",
      "      copying skbio\\stats\\ordination\\tests\\test_canonical_correspondence_analysis.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\n",
      "      copying skbio\\stats\\ordination\\tests\\test_correspondence_analysis.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\n",
      "      copying skbio\\stats\\ordination\\tests\\test_ordination_results.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\n",
      "      copying skbio\\stats\\ordination\\tests\\test_principal_coordinate_analysis.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\n",
      "      copying skbio\\stats\\ordination\\tests\\test_redundancy_analysis.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\n",
      "      copying skbio\\stats\\ordination\\tests\\test_util.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\n",
      "      copying skbio\\stats\\ordination\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\table\\tests\n",
      "      copying skbio\\table\\tests\\test_base.py -> build\\lib.win-amd64-cpython-311\\skbio\\table\\tests\n",
      "      copying skbio\\table\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\table\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\tree\\tests\n",
      "      copying skbio\\tree\\tests\\test_majority_rule.py -> build\\lib.win-amd64-cpython-311\\skbio\\tree\\tests\n",
      "      copying skbio\\tree\\tests\\test_nj.py -> build\\lib.win-amd64-cpython-311\\skbio\\tree\\tests\n",
      "      copying skbio\\tree\\tests\\test_tree.py -> build\\lib.win-amd64-cpython-311\\skbio\\tree\\tests\n",
      "      copying skbio\\tree\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\tree\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\util\\tests\n",
      "      copying skbio\\util\\tests\\test_decorator.py -> build\\lib.win-amd64-cpython-311\\skbio\\util\\tests\n",
      "      copying skbio\\util\\tests\\test_misc.py -> build\\lib.win-amd64-cpython-311\\skbio\\util\\tests\n",
      "      copying skbio\\util\\tests\\test_plotting.py -> build\\lib.win-amd64-cpython-311\\skbio\\util\\tests\n",
      "      copying skbio\\util\\tests\\test_testing.py -> build\\lib.win-amd64-cpython-311\\skbio\\util\\tests\n",
      "      copying skbio\\util\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\skbio\\util\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\\tests\\data\\qiime-191-tt\n",
      "      copying skbio\\diversity\\alpha\\tests\\data\\qiime-191-tt\\faith-pd.txt -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\\tests\\data\\qiime-191-tt\n",
      "      copying skbio\\diversity\\alpha\\tests\\data\\qiime-191-tt\\otu-table.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\\tests\\data\\qiime-191-tt\n",
      "      copying skbio\\diversity\\alpha\\tests\\data\\qiime-191-tt\\README.md -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\\tests\\data\\qiime-191-tt\n",
      "      copying skbio\\diversity\\alpha\\tests\\data\\qiime-191-tt\\tree.nwk -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\alpha\\tests\\data\\qiime-191-tt\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\diversity\\beta\\tests\\data\\qiime-191-tt\n",
      "      copying skbio\\diversity\\beta\\tests\\data\\qiime-191-tt\\otu-table.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\beta\\tests\\data\\qiime-191-tt\n",
      "      copying skbio\\diversity\\beta\\tests\\data\\qiime-191-tt\\README.md -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\beta\\tests\\data\\qiime-191-tt\n",
      "      copying skbio\\diversity\\beta\\tests\\data\\qiime-191-tt\\tree.nwk -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\beta\\tests\\data\\qiime-191-tt\n",
      "      copying skbio\\diversity\\beta\\tests\\data\\qiime-191-tt\\unweighted_unifrac_dm.txt -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\beta\\tests\\data\\qiime-191-tt\n",
      "      copying skbio\\diversity\\beta\\tests\\data\\qiime-191-tt\\weighted_normalized_unifrac_dm.txt -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\beta\\tests\\data\\qiime-191-tt\n",
      "      copying skbio\\diversity\\beta\\tests\\data\\qiime-191-tt\\weighted_unifrac_dm.txt -> build\\lib.win-amd64-cpython-311\\skbio\\diversity\\beta\\tests\\data\\qiime-191-tt\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\embedding\\tests\\data\n",
      "      copying skbio\\embedding\\tests\\data\\embed1.txt.npy -> build\\lib.win-amd64-cpython-311\\skbio\\embedding\\tests\\data\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\io\\tests\\data\n",
      "      copying skbio\\io\\tests\\data\\big5_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\tests\\data\n",
      "      copying skbio\\io\\tests\\data\\big5_file.bz2 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\tests\\data\n",
      "      copying skbio\\io\\tests\\data\\big5_file.gz -> build\\lib.win-amd64-cpython-311\\skbio\\io\\tests\\data\n",
      "      copying skbio\\io\\tests\\data\\example_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\tests\\data\n",
      "      copying skbio\\io\\tests\\data\\example_file.bz2 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\tests\\data\n",
      "      copying skbio\\io\\tests\\data\\example_file.gz -> build\\lib.win-amd64-cpython-311\\skbio\\io\\tests\\data\n",
      "      copying skbio\\io\\tests\\data\\real_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\tests\\data\n",
      "      copying skbio\\io\\tests\\data\\real_file_2 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\tests\\data\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast6_custom_minimal -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast6_custom_mixed_nans -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast6_custom_multi_line -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast6_custom_single_line -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast6_default_multi_line -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast6_default_single_line -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast6_invalid_column_types -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast6_invalid_number_of_columns -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast6_invalid_type_in_column -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast7_custom_minimal -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast7_custom_mixed_nans -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast7_custom_multi_line -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast7_custom_single_line -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast7_default_multi_line -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast7_default_single_line -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast7_invalid_differing_fields -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast7_invalid_for_sniffer -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast7_invalid_for_sniffer_2 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast7_invalid_gibberish -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast7_invalid_no_data -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast7_invalid_too_many_columns -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\blast7_invalid_unrecognized_field -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\embed1.txt.npy -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\embed2.txt.npy -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\embl_constructed -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\embl_feature_level_record -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\embl_feature_level_record_no_FT -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\embl_multi_records -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\embl_single_record -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\embl_single_record_lower -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\embl_single_record_simple -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\embl_single_record_upper -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\embl_uniprot_record -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\embl_w_beginning_whitespace -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\empty -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_diff_ids.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_double_qual.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_double_seq.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_long_qual.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_no_qual.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_qual_del.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_qual_escape.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_qual_null.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_qual_space.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_qual_tab.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_qual_unit_sep.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_qual_vtab.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_short_qual.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_spaces.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_tabs.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_trunc_at_plus.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_trunc_at_qual.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_trunc_at_seq.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_trunc_in_plus.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_trunc_in_qual.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_trunc_in_seq.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\error_trunc_in_title.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_10_seqs -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_3_seqs_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_3_seqs_non_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_5_blanks_start_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_5_ws_lines_start_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_6_blanks_start_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_6_ws_lines_start_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_blanks_end_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_blank_lines_between_records -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_description_newline_replacement_empty_str -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_description_newline_replacement_multi_char -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_description_newline_replacement_none -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_id_whitespace_replacement_empty_str -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_id_whitespace_replacement_multi_char -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_id_whitespace_replacement_none -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_invalid_after_10_seqs -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_invalid_blank_line_after_header -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_invalid_blank_line_within_sequence -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_invalid_blank_sequence -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_invalid_legacy_format -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_invalid_missing_header -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_invalid_missing_seq_data_first -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_invalid_missing_seq_data_last -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_invalid_missing_seq_data_middle -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_invalid_whitespace_line_after_header -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_invalid_whitespace_only_line_within_sequence -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_invalid_whitespace_only_sequence -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_max_width_1 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_max_width_5 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_mixed_qual_scores -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_multi_seq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_multi_seq_roundtrip -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_prot_seqs_odd_labels -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_single_bio_seq_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_single_bio_seq_non_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_single_dna_seq_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_single_dna_seq_non_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_single_prot_seq_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_single_prot_seq_non_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_single_rna_seq_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_single_rna_seq_non_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_single_seq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_tabular_msa_different_type -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_ws_lines_between_records -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fasta_ws_lines_end_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_5_blanks_start_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_5_ws_lines_start_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_blank_lines -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_invalid_blank_after_header -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_invalid_blank_after_plus -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_invalid_blank_after_seq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_invalid_blank_in_seq_at_symbol -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_invalid_blank_within_qual -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_invalid_blank_within_seq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_invalid_missing_header -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_invalid_missing_seq_data -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_invalid_ws_line_after_header -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_invalid_ws_line_after_plus -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_invalid_ws_line_after_seq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_invalid_ws_line_within_qual -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_invalid_ws_line_within_seq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_multi_blank_between_records -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_multi_blank_end_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_multi_blank_start_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_multi_seq_sanger -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_multi_whitespace_stripping -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_multi_ws_lines_between_records -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_multi_ws_lines_end_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_multi_ws_lines_start_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_single_seq_illumina1.3 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_single_seq_illumina1.8 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_whitespace_only_lines -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_wrapping_as_illumina_no_description -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_wrapping_as_sanger_no_description -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_wrapping_original_sanger_no_description -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_writer_illumina1.3_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_writer_sanger_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\fastq_writer_sanger_non_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\genbank_5_blanks_start_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\genbank_6_blanks_start_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\genbank_missing_locus_name -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\genbank_multi_records -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\genbank_single_record -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\genbank_single_record_lower -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\genbank_single_record_upper -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\genbank_w_beginning_whitespace -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\gff3_bad_missing_directive -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\gff3_bad_wrong_columns -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\gff3_dna -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\gff3_multi_record -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\gff3_single_record -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\illumina_full_range_as_illumina.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\illumina_full_range_as_sanger.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\illumina_full_range_original_illumina.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\legacy9_and_blast7_default -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\legacy9_invalid_differing_fields -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\legacy9_invalid_too_many_columns -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\legacy9_mixed_nans -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\legacy9_multi_line -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\legacy9_single_line -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\longreads_as_illumina.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\longreads_as_sanger.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\longreads_original_sanger.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\misc_dna_as_illumina.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\misc_dna_as_sanger.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\misc_dna_original_sanger.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\misc_rna_as_illumina.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\misc_rna_as_sanger.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\misc_rna_original_sanger.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error1 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error10 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error11 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error12 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error13 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error14 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error15 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error16 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error17 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error18 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error19 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error2 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error20 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error21 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error22 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error23 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error24 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error3 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error4 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error5 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error6 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error7 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error8 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_error9 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_example2_scores -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_example3_scores -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_exp_Ordination_CCA_site -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_exp_Ordination_CCA_site_constraints -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_exp_Ordination_CCA_species -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_exp_Ordination_PCoA_site -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_exp_Ordination_RDA_site -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_exp_Ordination_RDA_site_constraints -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_exp_Ordination_RDA_species -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_L&L_CA_data_scores -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\ordination_PCoA_sample_data_3_scores -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_dna_3_seqs -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_invalid_empty_line_after_header -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_invalid_empty_line_before_header -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_invalid_empty_line_between_seqs -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_invalid_header_too_long -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_invalid_header_too_short -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_invalid_no_header -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_invalid_seq_too_long -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_invalid_seq_too_short -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_invalid_too_few_seqs -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_invalid_too_many_seqs -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_invalid_zero_seqs -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_invalid_zero_seq_len -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_single_seq_long -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_single_seq_short -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_two_chunks -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_variable_length_ids -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_varied_whitespace_in_seqs -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_whitespace_in_header_1 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_whitespace_in_header_2 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\phylip_whitespace_in_header_3 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\prot.emb -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\prot_vec.emb -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qseq_invalid_filter -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qseq_invalid_lane -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qseq_invalid_read -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qseq_invalid_tile -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qseq_invalid_x -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qseq_invalid_y -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qseq_multi_seq_illumina1.3 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qseq_single_seq_sanger -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_2_seqs_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_3_seqs_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_3_seqs_defaults_desc_mismatch -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_3_seqs_defaults_extra -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_3_seqs_defaults_id_mismatch -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_3_seqs_defaults_length_mismatch -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_3_seqs_non_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_5_blanks_start_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_5_ws_lines_start_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_6_blanks_start_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_6_ws_lines_start_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_blanks_end_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_blank_lines_between_records -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_description_newline_replacement_empty_str -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_description_newline_replacement_multi_char -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_description_newline_replacement_none -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_id_whitespace_replacement_empty_str -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_id_whitespace_replacement_multi_char -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_id_whitespace_replacement_none -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_invalid_blank_line_after_header -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_invalid_blank_line_within_seq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_invalid_blank_sequence -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_invalid_legacy_format -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_invalid_missing_header -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_invalid_missing_qual_scores_first -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_invalid_missing_qual_scores_last -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_invalid_missing_qual_scores_middle -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_invalid_qual_scores_float -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_invalid_qual_scores_negative -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_invalid_qual_scores_over_255 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_invalid_qual_scores_string -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_invalid_whitespace_line_in_seq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_invalid_whitespace_only_sequence -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_invalid_ws_line_after_header -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_max_width_1 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_max_width_5 -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_multi_seq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_multi_seq_roundtrip -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_prot_seqs_odd_labels -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_single_bio_seq_non_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_single_dna_seq_non_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_single_nuc_seq_non_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_single_prot_seq_non_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_single_rna_seq_non_defaults -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_single_seq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_tabular_msa_different_type -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_ws_lines_between_records -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\qual_ws_lines_end_of_file -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\sample-metadata-comments-comment-char-id.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\sample-metadata-comments-mixed-case.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\sample-metadata-comments.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\sample-metadata-complete-types-directive.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\sample-metadata-empty-rows.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\sample-metadata-leading-trailing-whitespace-split-id.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\sample-metadata-leading-trailing-whitespace.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\sanger_full_range_as_illumina.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\sanger_full_range_as_sanger.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\sanger_full_range_original_sanger.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\solexa_full_range_original_solexa.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_all_data_types -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_blank_lines -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_data_only -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_different_padding -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_differing_gc_data_length -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_differing_gr_data_length -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_differing_seq_lengths -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_duplicate_gc -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_duplicate_gr -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_duplicate_sequence_names -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_duplicate_tree_ids -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_extensive -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_extensive_mixed -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_invalid_data_type -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_invalid_nonexistent_gr -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_invalid_nonexistent_gs -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_malformed_data_line -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_malformed_gc_line -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_malformed_gf_line -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_malformed_gr_line -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_malformed_gs_line -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_metadata_only -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_minimal -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_missing_footer -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_missing_header -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_missing_reference_items -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_missing_rn_tag -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_multiple_msa -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_multiple_multi_line_trees -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_multiple_references -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_multiple_trees -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_multi_line_tree_no_id -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_multi_line_tree_with_id -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_nonstring_labels -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_no_data -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_rna -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_runon_gf_no_whitespace -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_runon_gf_with_whitespace -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_runon_gs_no_whitespace -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_runon_gs_with_whitespace -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_runon_references -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_runon_references_mixed -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_single_reference -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_single_tree_without_id -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_single_tree_with_id -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_two_of_each_metadata -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\stockholm_whitespace_only_lines -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\taxdump_names.dmp -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\taxdump_nodes.dmp -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\tsv_10_fields -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\tsv_8_fields -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\whitespace_only -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\wrapping_as_illumina.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\wrapping_as_sanger.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      copying skbio\\io\\format\\tests\\data\\wrapping_original_sanger.fastq -> build\\lib.win-amd64-cpython-311\\skbio\\io\\format\\tests\\data\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\column-name-conflicts-with-id-header.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\comments-and-empty-rows-only.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\data-longer-than-header.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\directive-after-directives-section.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\directive-before-header.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\directive-longer-than-header.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\duplicate-column-names-with-whitespace.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\duplicate-column-names.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\duplicate-directives.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\duplicate-ids-with-whitespace.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\duplicate-ids.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\empty-column-name.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\empty-file -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\empty-id.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\header-only-with-comments-and-empty-rows.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\header-only.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\id-conflicts-with-id-header.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\invalid-header.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\missing-unknown-scheme.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\non-utf-8.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\qiime1-empty.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\simple-utf-16be.txt -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\simple-utf-16le.txt -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\types-directive-non-numeric.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\unrecognized-column-type.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\unrecognized-directive.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\whitespace-only-column-name.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      copying skbio\\metadata\\tests\\data\\invalid\\whitespace-only-id.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\invalid\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\all-cells-padded.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\biom-observation-metadata.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\BOM-simple.txt -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\case-insensitive-types-directive.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\column-order.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\comments.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\complete-types-directive.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\complete-types-directive_w_sk.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\empty-rows.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\empty-types-directive.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\jagged-trailing-columns.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\jagged.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\leading-trailing-whitespace.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\mac-line-endings.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\minimal.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\missing-data.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\missing-insdc-no-directive.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\missing-insdc.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\no-columns.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\no-id-or-column-name-type-cast.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\no-newline-at-eof.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\non-standard-characters.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\numeric-column.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\override-insdc.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\partial-types-directive.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\qiime1.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\qiita-preparation-information.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\qiita-sample-information.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\recommended-ids.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\rows-shorter-than-header.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\simple-with-directive.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\simple.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\simple.txt -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\single-column.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\single-id.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\trailing-columns.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\unix-line-endings.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      copying skbio\\metadata\\tests\\data\\valid\\windows-line-endings.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\metadata\\tests\\data\\valid\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\\data\n",
      "      copying skbio\\stats\\tests\\data\\cr_data_out -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\\data\n",
      "      copying skbio\\stats\\tests\\data\\cr_data_raw -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\\data\n",
      "      copying skbio\\stats\\tests\\data\\cr_no_data_out -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\\data\n",
      "      copying skbio\\stats\\tests\\data\\cr_no_data_raw -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\\data\n",
      "      copying skbio\\stats\\tests\\data\\gr_wo_msg_out -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\\data\n",
      "      copying skbio\\stats\\tests\\data\\gr_wo_msg_raw -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\\data\n",
      "      copying skbio\\stats\\tests\\data\\gr_w_msg_out -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\\data\n",
      "      copying skbio\\stats\\tests\\data\\gr_w_msg_raw -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\\data\n",
      "      copying skbio\\stats\\tests\\data\\vr_out -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\\data\n",
      "      copying skbio\\stats\\tests\\data\\vr_raw -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\\data\n",
      "      copying skbio\\stats\\tests\\data\\vr_real_out -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\\data\n",
      "      copying skbio\\stats\\tests\\data\\vr_real_raw -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\tests\\data\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\bioenv_df_vegan.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\bioenv_dm_vegan.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\bioenv_exp_results_vegan.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\df.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\df_extra_column.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\dm.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\dm2.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\dm3.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\dm4.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\dm_reordered.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\exp_results.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\exp_results_different_column_order.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\exp_results_single_column.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\frameSeries_dm.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\frameSeries_grouping.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\mantel_env_dm_vegan.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\mantel_veg_dm_vegan.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\moving_pictures_dm.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\moving_pictures_mf.tsv -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\pwmantel_exp_results_all_dms.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\pwmantel_exp_results_dm_dm2.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\pwmantel_exp_results_duplicate_dms.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\pwmantel_exp_results_minimal.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\pwmantel_exp_results_minimal_with_labels.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\pwmantel_exp_results_na_p_value.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      copying skbio\\stats\\distance\\tests\\data\\pwmantel_exp_results_reordered_distance_matrices.txt -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\distance\\tests\\data\n",
      "      creating build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example2_biplot_scaling1 -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example2_biplot_scaling2 -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example2_sample_constraints_scaling1 -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example2_sample_constraints_scaling2 -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example2_site_scaling1_from_vegan -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example2_site_scaling2_from_vegan -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example2_species_scaling1_from_vegan -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example2_species_scaling2_from_vegan -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example2_X -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example2_Y -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example3_biplot_scaling1 -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example3_biplot_scaling2 -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example3_sample_constraints_scaling1 -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example3_sample_constraints_scaling2 -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example3_site_scaling1_from_vegan -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example3_site_scaling2_from_vegan -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example3_species_scaling1_from_vegan -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example3_species_scaling2_from_vegan -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example3_X -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\example3_Y -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\exp_PCoAEigenResults_site -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\exp_PCoAzeros_site -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\L&L_CA_data -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\PCoA_biplot_descriptors -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\PCoA_biplot_projected_descriptors -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\PCoA_sample_data -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\PCoA_sample_data_12dim -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\PCoA_sample_data_2 -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\PCoA_sample_data_3 -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\PCoA_sample_data_3_eigh_ref_3dim -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\PCoA_sample_data_3_fsvd_ref_3dim -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\PCoA_skbio -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\varechem.csv -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\varespec.csv -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\vare_rda_biplot_from_vegan.csv -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\vare_rda_constraints_from_vegan.csv -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\vare_rda_eigvals_from_vegan.csv -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\vare_rda_propexpl_from_vegan.csv -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\vare_rda_sites_from_vegan.csv -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      copying skbio\\stats\\ordination\\tests\\data\\vare_rda_species_from_vegan.csv -> build\\lib.win-amd64-cpython-311\\skbio\\stats\\ordination\\tests\\data\n",
      "      running build_ext\n",
      "      building 'skbio.metadata._intersection' extension\n",
      "      error: Microsoft Visual C++ 14.0 or greater is required. Get it with \"Microsoft C++ Build Tools\": https://visualstudio.microsoft.com/visual-cpp-build-tools/\n",
      "      [end of output]\n",
      "  \n",
      "  note: This error originates from a subprocess, and is likely not a problem with pip.\n",
      "  ERROR: Failed building wheel for scikit-bio\n",
      "  error: subprocess-exited-with-error\n",
      "  \n",
      "  × Building wheel for biom-format (pyproject.toml) did not run successfully.\n",
      "  │ exit code: 1\n",
      "  ╰─> [332 lines of output]\n",
      "      <string>:15: SetuptoolsDeprecationWarning: The test command is disabled and references to it are deprecated.\n",
      "      !!\n",
      "      \n",
      "              ********************************************************************************\n",
      "              Please remove any references to `setuptools.command.test` in all supported versions of the affected package.\n",
      "      \n",
      "              This deprecation is overdue, please update your project and remove deprecated\n",
      "              calls to avoid build errors in the future.\n",
      "              ********************************************************************************\n",
      "      \n",
      "      !!\n",
      "      C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\setuptools\\_distutils\\dist.py:261: UserWarning: Unknown distribution option: 'tests_require'\n",
      "        warnings.warn(msg)\n",
      "      running bdist_wheel\n",
      "      running build\n",
      "      running build_py\n",
      "      creating build\\lib.win-amd64-cpython-311\\biom\n",
      "      copying biom\\err.py -> build\\lib.win-amd64-cpython-311\\biom\n",
      "      copying biom\\exception.py -> build\\lib.win-amd64-cpython-311\\biom\n",
      "      copying biom\\parse.py -> build\\lib.win-amd64-cpython-311\\biom\n",
      "      copying biom\\table.py -> build\\lib.win-amd64-cpython-311\\biom\n",
      "      copying biom\\util.py -> build\\lib.win-amd64-cpython-311\\biom\n",
      "      copying biom\\__init__.py -> build\\lib.win-amd64-cpython-311\\biom\n",
      "      creating build\\lib.win-amd64-cpython-311\\biom\\cli\n",
      "      copying biom\\cli\\installation_informer.py -> build\\lib.win-amd64-cpython-311\\biom\\cli\n",
      "      copying biom\\cli\\metadata_adder.py -> build\\lib.win-amd64-cpython-311\\biom\\cli\n",
      "      copying biom\\cli\\metadata_exporter.py -> build\\lib.win-amd64-cpython-311\\biom\\cli\n",
      "      copying biom\\cli\\table_converter.py -> build\\lib.win-amd64-cpython-311\\biom\\cli\n",
      "      copying biom\\cli\\table_head.py -> build\\lib.win-amd64-cpython-311\\biom\\cli\n",
      "      copying biom\\cli\\table_ids.py -> build\\lib.win-amd64-cpython-311\\biom\\cli\n",
      "      copying biom\\cli\\table_normalizer.py -> build\\lib.win-amd64-cpython-311\\biom\\cli\n",
      "      copying biom\\cli\\table_subsetter.py -> build\\lib.win-amd64-cpython-311\\biom\\cli\n",
      "      copying biom\\cli\\table_summarizer.py -> build\\lib.win-amd64-cpython-311\\biom\\cli\n",
      "      copying biom\\cli\\table_validator.py -> build\\lib.win-amd64-cpython-311\\biom\\cli\n",
      "      copying biom\\cli\\uc_processor.py -> build\\lib.win-amd64-cpython-311\\biom\\cli\n",
      "      copying biom\\cli\\util.py -> build\\lib.win-amd64-cpython-311\\biom\\cli\n",
      "      copying biom\\cli\\__init__.py -> build\\lib.win-amd64-cpython-311\\biom\\cli\n",
      "      creating build\\lib.win-amd64-cpython-311\\biom\\tests\n",
      "      copying biom\\tests\\long_lines.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\n",
      "      copying biom\\tests\\test_err.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\n",
      "      copying biom\\tests\\test_parse.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\n",
      "      copying biom\\tests\\test_table.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\n",
      "      copying biom\\tests\\test_util.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\n",
      "      copying biom\\tests\\__init__.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\n",
      "      creating build\\lib.win-amd64-cpython-311\\biom\\tests\\test_cli\n",
      "      copying biom\\tests\\test_cli\\test_add_metadata.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_cli\n",
      "      copying biom\\tests\\test_cli\\test_show_install_info.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_cli\n",
      "      copying biom\\tests\\test_cli\\test_subset_table.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_cli\n",
      "      copying biom\\tests\\test_cli\\test_summarize_table.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_cli\n",
      "      copying biom\\tests\\test_cli\\test_table_converter.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_cli\n",
      "      copying biom\\tests\\test_cli\\test_table_normalizer.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_cli\n",
      "      copying biom\\tests\\test_cli\\test_uc_processor.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_cli\n",
      "      copying biom\\tests\\test_cli\\test_validate_table.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_cli\n",
      "      copying biom\\tests\\test_cli\\__init__.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_cli\n",
      "      creating build\\lib.win-amd64-cpython-311\\biom\\tests\\test_data\n",
      "      copying biom\\tests\\test_data\\__init__.py -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_data\n",
      "      running egg_info\n",
      "      writing biom_format.egg-info\\PKG-INFO\n",
      "      writing dependency_links to biom_format.egg-info\\dependency_links.txt\n",
      "      writing entry points to biom_format.egg-info\\entry_points.txt\n",
      "      writing requirements to biom_format.egg-info\\requires.txt\n",
      "      writing top-level names to biom_format.egg-info\\top_level.txt\n",
      "      dependency C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\arrayobject.h won't be automatically included in the manifest: the path must be relative\n",
      "      dependency C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\arrayscalars.h won't be automatically included in the manifest: the path must be relative\n",
      "      dependency C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\ndarrayobject.h won't be automatically included in the manifest: the path must be relative\n",
      "      dependency C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\ndarraytypes.h won't be automatically included in the manifest: the path must be relative\n",
      "      dependency C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\ufuncobject.h won't be automatically included in the manifest: the path must be relative\n",
      "      dependency C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\arrayobject.h won't be automatically included in the manifest: the path must be relative\n",
      "      dependency C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\arrayscalars.h won't be automatically included in the manifest: the path must be relative\n",
      "      dependency C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\ndarrayobject.h won't be automatically included in the manifest: the path must be relative\n",
      "      dependency C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\ndarraytypes.h won't be automatically included in the manifest: the path must be relative\n",
      "      dependency C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\ufuncobject.h won't be automatically included in the manifest: the path must be relative\n",
      "      dependency C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\arrayobject.h won't be automatically included in the manifest: the path must be relative\n",
      "      dependency C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\arrayscalars.h won't be automatically included in the manifest: the path must be relative\n",
      "      dependency C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\ndarrayobject.h won't be automatically included in the manifest: the path must be relative\n",
      "      dependency C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\ndarraytypes.h won't be automatically included in the manifest: the path must be relative\n",
      "      dependency C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\ufuncobject.h won't be automatically included in the manifest: the path must be relative\n",
      "      reading manifest file 'biom_format.egg-info\\SOURCES.txt'\n",
      "      reading manifest template 'MANIFEST.in'\n",
      "      warning: no directories found matching 'support_files'\n",
      "      warning: no directories found matching 'tests\\test_data'\n",
      "      no previously-included directories found matching 'docs\\_build'\n",
      "      warning: no previously-included files matching '*.pyc' found anywhere in distribution\n",
      "      warning: no previously-included files matching '*.pyo' found anywhere in distribution\n",
      "      warning: no previously-included files matching '.git' found anywhere in distribution\n",
      "      warning: no previously-included files matching '*.so' found anywhere in distribution\n",
      "      warning: no previously-included files matching '.*.swp' found anywhere in distribution\n",
      "      adding license file 'COPYING.txt'\n",
      "      writing manifest file 'biom_format.egg-info\\SOURCES.txt'\n",
      "      C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\setuptools\\command\\build_py.py:212: _Warning: Package 'biom.assets' is absent from the `packages` configuration.\n",
      "      !!\n",
      "      \n",
      "              ********************************************************************************\n",
      "              ############################\n",
      "              # Package would be ignored #\n",
      "              ############################\n",
      "              Python recognizes 'biom.assets' as an importable package[^1],\n",
      "              but it is absent from setuptools' `packages` configuration.\n",
      "      \n",
      "              This leads to an ambiguous overall configuration. If you want to distribute this\n",
      "              package, please make sure that 'biom.assets' is explicitly added\n",
      "              to the `packages` configuration field.\n",
      "      \n",
      "              Alternatively, you can also rely on setuptools' discovery methods\n",
      "              (for example by using `find_namespace_packages(...)`/`find_namespace:`\n",
      "              instead of `find_packages(...)`/`find:`).\n",
      "      \n",
      "              You can read more about \"package discovery\" on setuptools documentation page:\n",
      "      \n",
      "              - https://setuptools.pypa.io/en/latest/userguide/package_discovery.html\n",
      "      \n",
      "              If you don't want 'biom.assets' to be distributed and are\n",
      "              already explicitly excluding 'biom.assets' via\n",
      "              `find_namespace_packages(...)/find_namespace` or `find_packages(...)/find`,\n",
      "              you can try to use `exclude_package_data`, or `include-package-data=False` in\n",
      "              combination with a more fine grained `package-data` configuration.\n",
      "      \n",
      "              You can read more about \"package data files\" on setuptools documentation page:\n",
      "      \n",
      "              - https://setuptools.pypa.io/en/latest/userguide/datafiles.html\n",
      "      \n",
      "      \n",
      "              [^1]: For Python, any directory (with suitable naming) can be imported,\n",
      "                    even if it does not contain any `.py` files.\n",
      "                    On the other hand, currently there is no concept of package data\n",
      "                    directory, all directories are treated like packages.\n",
      "              ********************************************************************************\n",
      "      \n",
      "      !!\n",
      "        check.warn(importable)\n",
      "      C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\setuptools\\command\\build_py.py:212: _Warning: Package 'biom.tests.bench_tables' is absent from the `packages` configuration.\n",
      "      !!\n",
      "      \n",
      "              ********************************************************************************\n",
      "              ############################\n",
      "              # Package would be ignored #\n",
      "              ############################\n",
      "              Python recognizes 'biom.tests.bench_tables' as an importable package[^1],\n",
      "              but it is absent from setuptools' `packages` configuration.\n",
      "      \n",
      "              This leads to an ambiguous overall configuration. If you want to distribute this\n",
      "              package, please make sure that 'biom.tests.bench_tables' is explicitly added\n",
      "              to the `packages` configuration field.\n",
      "      \n",
      "              Alternatively, you can also rely on setuptools' discovery methods\n",
      "              (for example by using `find_namespace_packages(...)`/`find_namespace:`\n",
      "              instead of `find_packages(...)`/`find:`).\n",
      "      \n",
      "              You can read more about \"package discovery\" on setuptools documentation page:\n",
      "      \n",
      "              - https://setuptools.pypa.io/en/latest/userguide/package_discovery.html\n",
      "      \n",
      "              If you don't want 'biom.tests.bench_tables' to be distributed and are\n",
      "              already explicitly excluding 'biom.tests.bench_tables' via\n",
      "              `find_namespace_packages(...)/find_namespace` or `find_packages(...)/find`,\n",
      "              you can try to use `exclude_package_data`, or `include-package-data=False` in\n",
      "              combination with a more fine grained `package-data` configuration.\n",
      "      \n",
      "              You can read more about \"package data files\" on setuptools documentation page:\n",
      "      \n",
      "              - https://setuptools.pypa.io/en/latest/userguide/datafiles.html\n",
      "      \n",
      "      \n",
      "              [^1]: For Python, any directory (with suitable naming) can be imported,\n",
      "                    even if it does not contain any `.py` files.\n",
      "                    On the other hand, currently there is no concept of package data\n",
      "                    directory, all directories are treated like packages.\n",
      "              ********************************************************************************\n",
      "      \n",
      "      !!\n",
      "        check.warn(importable)\n",
      "      C:\\Users\\Utente\\AppData\\Local\\Temp\\pip-build-env-4_1b402o\\overlay\\Lib\\site-packages\\setuptools\\command\\build_py.py:212: _Warning: Package 'biom.tests.test_cli.test_data' is absent from the `packages` configuration.\n",
      "      !!\n",
      "      \n",
      "              ********************************************************************************\n",
      "              ############################\n",
      "              # Package would be ignored #\n",
      "              ############################\n",
      "              Python recognizes 'biom.tests.test_cli.test_data' as an importable package[^1],\n",
      "              but it is absent from setuptools' `packages` configuration.\n",
      "      \n",
      "              This leads to an ambiguous overall configuration. If you want to distribute this\n",
      "              package, please make sure that 'biom.tests.test_cli.test_data' is explicitly added\n",
      "              to the `packages` configuration field.\n",
      "      \n",
      "              Alternatively, you can also rely on setuptools' discovery methods\n",
      "              (for example by using `find_namespace_packages(...)`/`find_namespace:`\n",
      "              instead of `find_packages(...)`/`find:`).\n",
      "      \n",
      "              You can read more about \"package discovery\" on setuptools documentation page:\n",
      "      \n",
      "              - https://setuptools.pypa.io/en/latest/userguide/package_discovery.html\n",
      "      \n",
      "              If you don't want 'biom.tests.test_cli.test_data' to be distributed and are\n",
      "              already explicitly excluding 'biom.tests.test_cli.test_data' via\n",
      "              `find_namespace_packages(...)/find_namespace` or `find_packages(...)/find`,\n",
      "              you can try to use `exclude_package_data`, or `include-package-data=False` in\n",
      "              combination with a more fine grained `package-data` configuration.\n",
      "      \n",
      "              You can read more about \"package data files\" on setuptools documentation page:\n",
      "      \n",
      "              - https://setuptools.pypa.io/en/latest/userguide/datafiles.html\n",
      "      \n",
      "      \n",
      "              [^1]: For Python, any directory (with suitable naming) can be imported,\n",
      "                    even if it does not contain any `.py` files.\n",
      "                    On the other hand, currently there is no concept of package data\n",
      "                    directory, all directories are treated like packages.\n",
      "              ********************************************************************************\n",
      "      \n",
      "      !!\n",
      "        check.warn(importable)\n",
      "      copying biom\\_filter.pyx -> build\\lib.win-amd64-cpython-311\\biom\n",
      "      copying biom\\_subsample.pyx -> build\\lib.win-amd64-cpython-311\\biom\n",
      "      copying biom\\_transform.pyx -> build\\lib.win-amd64-cpython-311\\biom\n",
      "      creating build\\lib.win-amd64-cpython-311\\biom\\assets\n",
      "      copying biom\\assets\\exercise_api.py -> build\\lib.win-amd64-cpython-311\\biom\\assets\n",
      "      copying biom\\assets\\exercise_cli.sh -> build\\lib.win-amd64-cpython-311\\biom\\assets\n",
      "      creating build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x10000x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x10000x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x10000x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x10000x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x10000x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x10000x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x1000x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x1000x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x1000x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x1000x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x1000x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x1000x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x100x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x100x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x100x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x100x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x100x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x100x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x10x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x10x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x10x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x10x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x10x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10000x10x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x10000x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x10000x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x10000x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x10000x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x10000x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x10000x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x1000x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x1000x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x1000x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x1000x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x1000x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x1000x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x100x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x100x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x100x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x100x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x100x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x100x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x10x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x10x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x10x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x10x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x10x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\1000x10x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x10000x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x10000x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x10000x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x10000x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x10000x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x10000x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x1000x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x1000x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x1000x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x1000x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x1000x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x1000x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x100x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x100x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x100x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x100x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x100x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x100x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x10x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x10x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x10x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x10x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x10x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\100x10x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x10000x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x10000x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x10000x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x10000x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x10000x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x10000x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x1000x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x1000x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x1000x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x1000x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x1000x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x1000x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x100x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x100x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x100x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x100x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x100x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x100x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x10x0.001_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x10x0.001_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x10x0.010_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x10x0.010_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x10x0.100_bench.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      copying biom\\tests\\bench_tables\\10x10x0.100_bench_hdf5.biom.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\bench_tables\n",
      "      creating build\\lib.win-amd64-cpython-311\\biom\\tests\\test_cli\\test_data\n",
      "      copying biom\\tests\\test_cli\\test_data\\json_obs_collapsed.biom -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_cli\\test_data\n",
      "      copying biom\\tests\\test_cli\\test_data\\json_sample_collapsed.biom -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_cli\\test_data\n",
      "      copying biom\\tests\\test_cli\\test_data\\test.biom -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_cli\\test_data\n",
      "      copying biom\\tests\\test_cli\\test_data\\test.json -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_cli\\test_data\n",
      "      copying biom\\tests\\test_data\\bad_table.txt -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_data\n",
      "      copying biom\\tests\\test_data\\bad_table.txt.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_data\n",
      "      copying biom\\tests\\test_data\\edgecase_issue_952.biom -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_data\n",
      "      copying biom\\tests\\test_data\\empty.biom -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_data\n",
      "      copying biom\\tests\\test_data\\no-contents.biom -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_data\n",
      "      copying biom\\tests\\test_data\\test.biom -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_data\n",
      "      copying biom\\tests\\test_data\\test.json -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_data\n",
      "      copying biom\\tests\\test_data\\test.json.gz -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_data\n",
      "      copying biom\\tests\\test_data\\test_grp_metadata.biom -> build\\lib.win-amd64-cpython-311\\biom\\tests\\test_data\n",
      "      running build_ext\n",
      "      building 'biom._filter' extension\n",
      "      error: Microsoft Visual C++ 14.0 or greater is required. Get it with \"Microsoft C++ Build Tools\": https://visualstudio.microsoft.com/visual-cpp-build-tools/\n",
      "      [end of output]\n",
      "  \n",
      "  note: This error originates from a subprocess, and is likely not a problem with pip.\n",
      "  ERROR: Failed building wheel for biom-format\n",
      "ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (scikit-bio, biom-format)\n"
     ]
    }
   ],
   "source": [
    "!pip install scikit-bio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OTU</th>\n",
       "      <th>taxonomy</th>\n",
       "      <th>A1</th>\n",
       "      <th>A10</th>\n",
       "      <th>A100</th>\n",
       "      <th>A101</th>\n",
       "      <th>A102</th>\n",
       "      <th>A104</th>\n",
       "      <th>A105</th>\n",
       "      <th>A106</th>\n",
       "      <th>...</th>\n",
       "      <th>B52</th>\n",
       "      <th>B54</th>\n",
       "      <th>B55</th>\n",
       "      <th>B56</th>\n",
       "      <th>B57</th>\n",
       "      <th>B58</th>\n",
       "      <th>B59</th>\n",
       "      <th>B6</th>\n",
       "      <th>B60</th>\n",
       "      <th>B61</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OTU1</td>\n",
       "      <td>d__Bacteria;_k__norank;_p__Firmicutes;_c__Clos...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OTU2</td>\n",
       "      <td>d__Bacteria;_k__norank;_p__Proteobacteria;_c__...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OTU3</td>\n",
       "      <td>d__Bacteria;_k__norank;_p__Firmicutes;_c__Erys...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OTU4</td>\n",
       "      <td>d__Bacteria;_k__norank;_p__Firmicutes;_c__Baci...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OTU5</td>\n",
       "      <td>d__Bacteria;_k__norank;_p__Tenericutes;_c__Mol...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 256 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    OTU                                           taxonomy  A1  A10  A100  \\\n",
       "0  OTU1  d__Bacteria;_k__norank;_p__Firmicutes;_c__Clos...   0    0     0   \n",
       "1  OTU2  d__Bacteria;_k__norank;_p__Proteobacteria;_c__...   0    0     0   \n",
       "2  OTU3  d__Bacteria;_k__norank;_p__Firmicutes;_c__Erys...   0    0     0   \n",
       "3  OTU4  d__Bacteria;_k__norank;_p__Firmicutes;_c__Baci...   0    0     0   \n",
       "4  OTU5  d__Bacteria;_k__norank;_p__Tenericutes;_c__Mol...   0    0     1   \n",
       "\n",
       "   A101  A102  A104  A105  A106  ...  B52  B54  B55  B56  B57  B58  B59  B6  \\\n",
       "0     0     0     0     1     0  ...    0    0    0    0    0    0    0   0   \n",
       "1     0     0     0     0     0  ...    0    0    0    0    0    0    0   0   \n",
       "2     0     0     0     0     2  ...    0    0    0    0    0    0    0   0   \n",
       "3     0     0     0     0     0  ...    0    0    0    0    0    0    0   0   \n",
       "4     0     0     1     0     3  ...    0    0    0    0    0    0    0   0   \n",
       "\n",
       "   B60  B61  \n",
       "0    0    0  \n",
       "1    0    0  \n",
       "2    0    0  \n",
       "3    1    0  \n",
       "4    0    0  \n",
       "\n",
       "[5 rows x 256 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "\n",
    "# Load the dataset\n",
    "genus = pd.read_csv(r\"C:\\Users\\Utente\\Desktop\\SAE-microbiome\\GSE113690_Autism_16S_rRNA_OTU_assignment_and_abundance.csv\")\n",
    "\n",
    "# Display the dataset\n",
    "genus.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['A1', 'A10', 'A100', 'A101', 'A102', 'A104', 'A105', 'A106', 'A108',\n",
      "       'A109',\n",
      "       ...\n",
      "       'B52', 'B54', 'B55', 'B56', 'B57', 'B58', 'B59', 'B6', 'B60', 'B61'],\n",
      "      dtype='object', length=254)\n"
     ]
    }
   ],
   "source": [
    "# Identify columns with all zero values\n",
    "zero_cols = genus.columns[(genus == 0).any()]\n",
    "print(zero_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>OTU</th>\n",
       "      <th>OTU1</th>\n",
       "      <th>OTU2</th>\n",
       "      <th>OTU3</th>\n",
       "      <th>OTU4</th>\n",
       "      <th>OTU5</th>\n",
       "      <th>OTU6</th>\n",
       "      <th>OTU7</th>\n",
       "      <th>OTU8</th>\n",
       "      <th>OTU9</th>\n",
       "      <th>OTU10</th>\n",
       "      <th>...</th>\n",
       "      <th>OTU1329</th>\n",
       "      <th>OTU1330</th>\n",
       "      <th>OTU1331</th>\n",
       "      <th>OTU1332</th>\n",
       "      <th>OTU1333</th>\n",
       "      <th>OTU1334</th>\n",
       "      <th>OTU1335</th>\n",
       "      <th>OTU1336</th>\n",
       "      <th>OTU1337</th>\n",
       "      <th>OTU1338</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A10</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A100</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>485</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A101</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>142</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A102</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1322 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "OTU   OTU1  OTU2  OTU3  OTU4  OTU5  OTU6  OTU7  OTU8  OTU9  OTU10  ...  \\\n",
       "A1       0     0     0     0     0     1     0     0    50      0  ...   \n",
       "A10      0     0     0     0     0     0     0     0    10      2  ...   \n",
       "A100     0     0     0     0     1     0     0   485    13      0  ...   \n",
       "A101     0     0     0     0     0     0     1   142    17      0  ...   \n",
       "A102     0     0     0     0     0     0     0     1     9      0  ...   \n",
       "\n",
       "OTU   OTU1329  OTU1330  OTU1331  OTU1332  OTU1333  OTU1334  OTU1335  OTU1336  \\\n",
       "A1          1        0        1        0        0        0        0        0   \n",
       "A10         5        0        0        0        0        0        0        0   \n",
       "A100        0        0        0        0        0        0        0        0   \n",
       "A101        0        1        0        0        0        0        0        0   \n",
       "A102        1        0        0        0        0        0        0        0   \n",
       "\n",
       "OTU   OTU1337  OTU1338  \n",
       "A1          0        0  \n",
       "A10         0        0  \n",
       "A100        0        0  \n",
       "A101        0        0  \n",
       "A102        0        0  \n",
       "\n",
       "[5 rows x 1322 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Extract OTU and taxonomy information\n",
    "taxa = genus[['OTU', 'taxonomy']].set_index('OTU')\n",
    "# Transpose the dataset and set OTU as index\n",
    "genus_T = genus.drop('taxonomy', axis=1).set_index('OTU').transpose()\n",
    "genus_T.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,\n",
       "       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,\n",
       "       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,\n",
       "       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,\n",
       "       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,\n",
       "       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,\n",
       "       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
       "       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
       "       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
       "       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
       "       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
       "       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create binary target variable based on sample names\n",
    "target = genus_T.index.to_list()\n",
    "binary_target = np.array([1 if t.startswith('A') else 0 for t in target ])\n",
    "binary_target"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>OTU</th>\n",
       "      <th>OTU1</th>\n",
       "      <th>OTU2</th>\n",
       "      <th>OTU3</th>\n",
       "      <th>OTU4</th>\n",
       "      <th>OTU5</th>\n",
       "      <th>OTU6</th>\n",
       "      <th>OTU7</th>\n",
       "      <th>OTU8</th>\n",
       "      <th>OTU9</th>\n",
       "      <th>OTU10</th>\n",
       "      <th>...</th>\n",
       "      <th>OTU1330</th>\n",
       "      <th>OTU1331</th>\n",
       "      <th>OTU1332</th>\n",
       "      <th>OTU1333</th>\n",
       "      <th>OTU1334</th>\n",
       "      <th>OTU1335</th>\n",
       "      <th>OTU1336</th>\n",
       "      <th>OTU1337</th>\n",
       "      <th>OTU1338</th>\n",
       "      <th>Diagnosis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1</th>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00006</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0.001542</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00006</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A10</th>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0.000333</td>\n",
       "      <td>0.000091</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A100</th>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00006</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.014692</td>\n",
       "      <td>0.000423</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A101</th>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00006</td>\n",
       "      <td>0.004323</td>\n",
       "      <td>0.000544</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00006</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A102</th>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.000060</td>\n",
       "      <td>0.000302</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1323 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "OTU      OTU1     OTU2     OTU3     OTU4     OTU5     OTU6     OTU7      OTU8  \\\n",
       "A1    0.00003  0.00003  0.00003  0.00003  0.00003  0.00006  0.00003  0.000030   \n",
       "A10   0.00003  0.00003  0.00003  0.00003  0.00003  0.00003  0.00003  0.000030   \n",
       "A100  0.00003  0.00003  0.00003  0.00003  0.00006  0.00003  0.00003  0.014692   \n",
       "A101  0.00003  0.00003  0.00003  0.00003  0.00003  0.00003  0.00006  0.004323   \n",
       "A102  0.00003  0.00003  0.00003  0.00003  0.00003  0.00003  0.00003  0.000060   \n",
       "\n",
       "OTU       OTU9     OTU10  ...  OTU1330  OTU1331  OTU1332  OTU1333  OTU1334  \\\n",
       "A1    0.001542  0.000030  ...  0.00003  0.00006  0.00003  0.00003  0.00003   \n",
       "A10   0.000333  0.000091  ...  0.00003  0.00003  0.00003  0.00003  0.00003   \n",
       "A100  0.000423  0.000030  ...  0.00003  0.00003  0.00003  0.00003  0.00003   \n",
       "A101  0.000544  0.000030  ...  0.00006  0.00003  0.00003  0.00003  0.00003   \n",
       "A102  0.000302  0.000030  ...  0.00003  0.00003  0.00003  0.00003  0.00003   \n",
       "\n",
       "OTU   OTU1335  OTU1336  OTU1337  OTU1338  Diagnosis  \n",
       "A1    0.00003  0.00003  0.00003  0.00003          1  \n",
       "A10   0.00003  0.00003  0.00003  0.00003          1  \n",
       "A100  0.00003  0.00003  0.00003  0.00003          1  \n",
       "A101  0.00003  0.00003  0.00003  0.00003          1  \n",
       "A102  0.00003  0.00003  0.00003  0.00003          1  \n",
       "\n",
       "[5 rows x 1323 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create binary target variable based on sample names\n",
    "target = genus_T.index.to_list()\n",
    "binary_target = np.array([1 if t.startswith('A') else 0 for t in target ])\n",
    "\n",
    "# Add 1 to all values to avoid division by zero\n",
    "genus_T = genus_T + 1\n",
    "# Calculate total species count for each sample\n",
    "total_species = genus_T.sum(axis=1)\n",
    "# Define absolute abundance value\n",
    "abs_abundance = 31757\n",
    "# Calculate relative abundance\n",
    "pd_rel_abundance = genus_T.div(total_species, axis=0)\n",
    "\n",
    "# Add Diagnosis column to the relative abundance dataframe\n",
    "pd_rel_abundance['Diagnosis'] = binary_target\n",
    "\n",
    "pd_rel_abundance.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
